File: arkhe_multilang/rosetta/agent.ex
defmodule Agent do
  defstruct name: "", counter: 0
end

defmodule Handover do
  def send_message(sender, receiver) do
    %{receiver | counter: receiver.counter + 1}
  end
end

alice = %Agent{name: "Alice"}
bob = %Agent{name: "Bob"}
bob = Handover.send_message(alice, bob)
IO.inspect(bob.counter)

File: arkhe_multilang/rosetta/agent.go
package main

import "fmt"

type Agent struct {
    Name    string
    Counter int
}

func sendMessage(sender Agent, receiver *Agent) {
    receiver.Counter++
    fmt.Printf("%s -> %s: %d\n", sender.Name, receiver.Name, receiver.Counter)
}

func main() {
    alice := Agent{Name: "Alice"}
    bob := &Agent{Name: "Bob"}
    sendMessage(alice, bob)
    sendMessage(alice, bob)
}
